client/modules/driver/reducers/route.js

```diff
@@ -97,7 +97,12 @@ export const requestGoogleRoute = (waypoints, optimize) => dispatch => {
 
       dispatch(setWaypoints(optimizedWaypoints))
     })
-    .catch(err => dispatch(routeFailure(err)))
+    .catch(err => {
+      if (err.isDirectionsError)
+        dispatch(routeFailure(err.message))
+      else
+        throw err
```

Comment:
I think we should still `dispatch(routeFailure('There was a problem getting your route'))` or something here so it's visible to the user that something went wrong.

Reply:
The caught error is likely programmatic, and unless an invalid value was passed to `google.maps.DirectionsService.route()`, it has nothing to do with getting the route. It could be a component missing a prop as in PR #312.

But if you don't want to have the app fail in such situations, I can change this PR to just fix the try catch block and improve how direction errors are reported.

Reply:
There's also the `MAX_WAYPOINTS_EXCEEDED` and maybe `OVER_QUERY_LIMIT` errors that could be useful to show. If it's a problem in the app I still think it's nicer to have a 'you clicked this and we tried but something went wrong' indication rather than a button that seems to do nothing.

I don't think it needs the try catch in `getGoogleRoute` though, just make an error with `isDirectionsError` like it is now but with an appropriate message, then in the catch of the `requestGoogleRoute` action creator dispatch a `routeFailure` with a generic error message if it's not a `isDirectionsError`.